src/crewai/crew.py

```diff
@@ -779,10 +762,11 @@ def _log_task_start(self, task: Task, role: str = "None"):
             self._file_handler.log(agent=role, task=task.description, status="started")
 
     def _update_manager_tools(self, task: Task, manager: BaseAgent):
```

Comment:
Why not drop the manager parameter?

Also, why not change the else to this?

` self.manager_agent.tools = manager.get_delegation_tools(self.agents)`